### DISCUSSION TITLE
Search within categories

### DISCUSSION THREAD

**Discussion by zerenxyz** [squidfunk/mkdocs-material#7601] [Q&A] [answered]
I am uncertain whether the current version of MkDocs Material supports this feature, but currently, the search function queries across all files and pages. Is it possible to limit the search function to search within a specific page and its subpages only? Alternatively, is the best approach to create separate instances for each page I need and add them as links in the header? Or maybe using the Projects functionality from Insiders? 

For example:
```
.
├── category_a/
│   ├── topic_1/
│   │   ├── topic_1.1
│   │   └── ...
│   ├── topic_2/
│   │   ├── topic_2.1/
│   │   │   ├── topic_2.1.1
│   │   │   └── ...
│   │   └── ...
│   └── topic_3
├── category_b/
│   ├── topic_1/
│   │   ├── topic_1.1
│   │   └── ...
│   └── topic_2/
│       ├── topic_2.1/
│       │   ├── topic_2.1.1
│       │   └── ...
│       └── ...
└── category_c/
    └── topic_1/
        ├── topic_1.1
        └── ...
```
So if you are on the page Category A the search would only show results from the subpages of that category, and same thing for the rest of categories.

**Comment by squidfunk** [comment, marked as answer]
Yes, you can either use projects to create separate projects with distinct search indexes. Other than that, scoping searches to subtrees is already implemented in the new search that is coming, tracked in #6307. Please refer to this issue for updates, and no, we currently do not have a release date due to the changes we must tackle first, as mentioned in [our latest blog posts](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/).

**Reply by zerenxyz** [reply]
By using projects it will all live under one mkdocs material install but have their own search limited to the project scope as far as I understand it? I think that solves it then for now until the new search comes out, thank you for the quick response!

**Reply by squidfunk** [reply]
Yes, correct. You can check [our examples project](https://github.com/mkdocs-material/examples), with each example having a separate search index.
